src/main/java/edu/harvard/iq/dataverse/UserServiceBean.java

```diff
@@ -34,5 +41,368 @@ public AuthenticatedUser save( AuthenticatedUser user ) {
+        qstr += " u.affiliation, u.superuser,";
+        qstr += " u.position, u.modificationtime,";
+        qstr += " prov.id, prov.factoryalias";
+        qstr += " FROM authenticateduser u,";
+        qstr += " authenticateduserlookup prov_lookup,";
+        qstr += " authenticationproviderrow prov";
+        qstr += " WHERE";
+        qstr += " u.id = prov_lookup.authenticateduser_id";
+        qstr += " AND prov_lookup.authenticationproviderid = prov.id";       
+        qstr += sharedSearchClause;
+        qstr += " ORDER BY u.useridentifier";
+        qstr += " LIMIT " + resultLimit;
+        qstr += " OFFSET " + offset;
+        qstr += ";";
+        
+        System.out.println("getUserCount: " + qstr);
+
+        Query nativeQuery = em.createNativeQuery(qstr);           
+       
+        return nativeQuery.getResultList();
+
+    }
+    
+    /**
+     * The search clause needs to be consistent between the searches that:
+     * (1) get a user count
+     * (2) get a list of users
+     * 
+     * @param searchTerm
+     * @return 
+     */
+    private String getSharedSearchClause(String searchTerm){       
+        if (searchTerm == null){
+            return "";
+        }
+
+        String searchClause = "";
+
+        searchTerm = searchTerm.trim();
+        if (searchTerm.isEmpty()){
+            return "";
+        }        
+        
+        //--------------------------------------------------------
+        // Search for term anywhere at beginning of string
+        //--------------------------------------------------------
+        searchClause += " (u.useridentifier LIKE '" + searchTerm +"%'";
+        searchClause += " OR u.firstname ILIKE '" + searchTerm +"%'";
+        searchClause += " OR u.lastname ILIKE '" + searchTerm +"%'";
+        searchClause += " OR u.email ILIKE '" + searchTerm +"%')";
+        
+        /*
+        //--------------------------------------------------------
+        // Search for term anywhere in string
+        //--------------------------------------------------------
+        searchClause += " u.useridentifier LIKE '%" + searchTerm +"%'";
+        searchClause += " OR u.firstname ILIKE '%" + searchTerm +"%'";
+        searchClause += " OR u.lastname ILIKE '%" + searchTerm +"%'";
+        searchClause += " OR u.email ILIKE '%" + searchTerm +"%'";
+        */
+        return searchClause;
+    }
+    
+    
+    /**
+     * Return the number of superusers -- for the dashboard
+     * @return 
+     */
+    public Long getSuperUserCount() {
+        
+        String qstr = "SELECT count(id)";
+        qstr += " FROM authenticateduser";
+        qstr += " WHERE superuser = true";
+        qstr += ";";
+        
+        Query nativeQuery = em.createNativeQuery(qstr);  
+
+        return (Long)nativeQuery.getSingleResult();
 
+    }
+
+    /**
+     * Return count of all users
+     * @return 
+     */
+    public Long getTotalUserCount(){
+        
+        return getUserCount(null);
+    }
```

Comment:
As getUserCount immediate checks for null and changes it, why not pass an empty string instead?